### PR TITLE
Move bulk upload HIV beta behind new feature flag 

### PR DESCRIPTION
**Pull request by bobbywells52**
# BACKEND & FRONTEND PULL REQUEST

## Related Issue

https://github.com/CDCgov/prime-simplereport/issues/7228

## Changes Proposed 

- Create a `hivBulkUploadEnabled` flag and use it in place of the `hivEnabled` flag to gate if the hiv bulk upload Beta is accessible in dev envs.

## Additional Information

- The test labled, "loads menu categories including Beta" in `SupportAdmin.test.tsx` was removed as it tested against the old use case of the `hivEnabled` flag.
## Testing
I've deployed on dev6 and everything seems to be running properly.

Pull the branch down into your local env and run it
- [ ] Confirm that the admin support page loads with the Beta section
- [ ] Set `hivBulkUploadEnabled` to false in `backend/src/main/resources/application.yaml`
- [ ] Reload the admin support page and confirm that the Beta section no longer renders 
- [ ] Verify that `/admin/hiv-csv-upload` endpoint cannot be hit

### CODE REVIEW

**Review by fzhao99 — APPROVED** [review]
lgtm

**Review by emyl3 — CHANGES_REQUESTED** [review]
We also want to update the endpoint to use the new feature flag
https://github.com/CDCgov/prime-simplereport/blob/b551b21939b446fab87e785a5152301c1d68567e/backend/src/main/java/gov/cdc/usds/simplereport/api/uploads/FileUploadController.java#L54

nice work, bobby!

**Review by emyl3 — COMMENTED** [review]
I think we also need to update the FileUploadControllerTest file -- but I don't think the 
`when(featureFlagsConfig.isHivEnabled()).thenReturn(true);` lines serve a purpose because regardless the test passes... which I did not expect 👀

**Comment by bobbywells52** [comment]
> I think we also need to update the FileUploadControllerTest file -- but I don't think the `when(featureFlagsConfig.isHivEnabled()).thenReturn(true);` lines serve a purpose because regardless the test passes... which I did not expect 👀

Hmmm yeah I'm not quite sure if that line is doing anything at all. I've tried setting it to the four following permutations and every time the test has passed locally:

`when(featureFlagsConfig.isHivEnabled()).thenReturn(true);`
`when(featureFlagsConfig.isHivEnabled()).thenReturn(false);`
`when(featureFlagsConfig.isHivBulkUploadEnabled()).thenReturn(true);`
`when(featureFlagsConfig.isHivBulkUploadEnabled()).thenReturn(false);`

I've never working with the spring boot testing framework before and I'm not quite sure what is going on here 🤔

**Comment by bobbywells52** [comment]
> > I think we also need to update the FileUploadControllerTest file -- but I don't think the `when(featureFlagsConfig.isHivEnabled()).thenReturn(true);` lines serve a purpose because regardless the test passes... which I did not expect 👀
> 
> Hmmm yeah I'm not quite sure if that line is doing anything at all. I've tried setting it to the four following permutations and every time the test has passed locally:
> 
> `when(featureFlagsConfig.isHivEnabled()).thenReturn(true);` `when(featureFlagsConfig.isHivEnabled()).thenReturn(false);` `when(featureFlagsConfig.isHivBulkUploadEnabled()).thenReturn(true);` `when(featureFlagsConfig.isHivBulkUploadEnabled()).thenReturn(false);`
> 
> I've never worked with the spring boot testing framework before and I'm not quite sure what is going on here 🤔

@emyl3 paired together and agreed that having the test set against the `isHivBulkUploadEnabled` flag opposed to the `isHivEnabled` flag and passing suffices. However, we are both left with questions as to how the mocking for flags is working in this part of the app.

 Will add to docket to discuss at next eng sync.

**Review by emyl3 — DISMISSED** [review]
LGTM! Thank you Bobby!

File: backend/src/test/java/gov/cdc/usds/simplereport/api/uploads/FileUploadControllerTest.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ void handleHIVResultsUpload_successful() throws IOException {
             .warnings(empty)
             .destination(Pipeline.UNIVERSAL)
             .build();
-    when(featureFlagsConfig.isHivEnabled()).thenReturn(true);
+    when(featureFlagsConfig.isHivBulkUploadEnabled()).thenReturn(true);
```
Comment (fzhao99) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/uploads/FileUploadControllerTest.java:50]: Could we actually pull that test out and then add it back later once we figure out what the issue is? Would rather not have a test than have one that’s passing in a misleading way

Comment (fzhao99) [reply backend/src/test/java/gov/cdc/usds/simplereport/api/uploads/FileUploadControllerTest.java:50]: fysa @emyl3

Comment (bobbywells52) [reply backend/src/test/java/gov/cdc/usds/simplereport/api/uploads/FileUploadControllerTest.java]: New issue for the test here: https://github.com/CDCgov/prime-simplereport/issues/7240

**Review by fzhao99 — APPROVED** [review]
lgtm! good work on this :)

**Review by emyl3 — APPROVED** [review]
LGTM!

**Pull request merged**